src/ApiService/ApiService/Functions/InstanceConfig.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+
+namespace Microsoft.OneFuzz.Service.Functions;
+
+public class InstanceConfig {
+    private readonly ILogTracer _log;
+    private readonly IEndpointAuthorization _auth;
+    private readonly IOnefuzzContext _context;
+
+    public InstanceConfig(ILogTracer log, IEndpointAuthorization auth, IOnefuzzContext context) {
+        _log = log;
+        _auth = auth;
+        _context = context;
+    }
+
+    [Function("InstanceConfig")]
+    public Async.Task<HttpResponseData> Run(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "GET", "POST", Route = "instance_config")] HttpRequestData req) {
+        return _auth.CallIfUser(req, r => r.Method switch {
+            "GET" => Get(r),
+            "POST" => Post(r),
+            _ => throw new InvalidOperationException("Unsupported HTTP method"),
+        });
+    }
+    public async Async.Task<HttpResponseData> Get(HttpRequestData req) {
+        _log.Info($"getting instance_config");
+        var config = await _context.ConfigOperations.Fetch();
+
+        var response = req.CreateResponse(HttpStatusCode.OK);
+        await response.WriteAsJsonAsync(config);
+        return response;
+    }
+
+    public async Async.Task<HttpResponseData> Post(HttpRequestData req) {
+        _log.Info($"attempting instance_config update");
+        var request = await RequestHandling.ParseRequest<InstanceConfigUpdate>(req);
+
+        if (!request.IsOk) {
+            return await _context.RequestHandling.NotOk(
+                req,
+                request.ErrorV,
+                context: "instance_config update");
+        }
+        var config = await _context.ConfigOperations.Fetch();
+        var answer = await _auth.CheckRequireAdmins(req);
```

Comment:
You can await both to do these operations in parallel:

(also needs `using System.Threading.Tasks;` at top) 

```suggestion
        var (config, answer) = await (
            _context.ConfigOperations.Fetch(),
            _auth.CheckRequireAdmins(req));
```